File: webapi/Models/Flexibility/Tips/TipExercise.cs
using webapi.Models.Math;

namespace webapi.Models.Flexibility
{
    public class TipExercise : IFlexibilityExercise
    {
        public long? Id { get; set; }

        public int? Ordering { get; set; }

        public FlexibilityExerciseType Type { get; } = FlexibilityExerciseType.TipExercise;

        public Method Method { get; set; }

        public LinearEquation FirstEquation { get; set; }

        public LinearEquation SecondEquation { get; set; }

        public IsolatedIn FirstEquationIsIsolatedIn { get; set; }

        public IsolatedIn SecondEquationIsIsolatedIn { get; set; }

        public Variable FirstVariable { get; set; }

        public Variable SecondVariable { get; set; }

        public string Question { get; set; }

        public string AgentMessageForTask { get; set; }

        public string AgentMessageForFirstSolution { get; set; }

        public string AgentMessageForSecondSolution { get; set; }

        public TipExercise() { }
    }
}

File: reactapp/src/types/flexibility/tipExercise.ts
import { Type } from "class-transformer";
import { FlexibilityEquation } from "@/types/math/linearEquation.ts";
import { Variable } from "@/types/flexibility/variable.ts";
import { IsolatedIn, Method } from "@/types/flexibility/enums.ts";
import { IFlexibilityExercise } from "@/types/flexibility/flexibilityExercise.ts";
import { FlexibilityStudyExerciseType } from "@/types/studies/enums.ts";

export class TipExercise implements IFlexibilityExercise {
    public readonly id: number;

    public readonly type: FlexibilityStudyExerciseType = FlexibilityStudyExerciseType.TipExercise;

    public readonly method: Method;

    @Type(() => FlexibilityEquation)
    public readonly firstEquation: FlexibilityEquation;

    @Type(() => FlexibilityEquation)
    public readonly secondEquation: FlexibilityEquation;

    public readonly firstEquationIsIsolatedIn: IsolatedIn;

    public readonly secondEquationIsIsolatedIn: IsolatedIn;

    @Type(() => Variable)
    public readonly firstVariable: Variable;

    @Type(() => Variable)
    public readonly secondVariable: Variable;

    public readonly question: string;

    public readonly agentMessageForTask?: string | undefined;

    public readonly agentMessageForFirstSolution?: string | undefined;

    public readonly agentMessageForSecondSolution?: string | undefined;

    constructor(id: number, method: Method, firstEquation: FlexibilityEquation, secondEquation: FlexibilityEquation, firstEquationIsIsolatedIn: IsolatedIn,
                secondEquationIsIsolatedIn: IsolatedIn, firstVariable: Variable, secondVariable: Variable, question: string,
                agentMessageForTask?: string | null, agentMessageForFirstSolution?: string | null, agentMessageForSecondSolution?: string | null) {
        this.id = id;
        this.method = method;
        this.firstEquation = firstEquation;
        this.secondEquation = secondEquation;
        this.firstEquationIsIsolatedIn = firstEquationIsIsolatedIn;
        this.secondEquationIsIsolatedIn = secondEquationIsIsolatedIn;
        this.firstVariable = firstVariable;
        this.secondVariable = secondVariable;
        this.question = question;
        this.agentMessageForTask = agentMessageForTask === null ? undefined : agentMessageForTask;
        this.agentMessageForFirstSolution = agentMessageForFirstSolution === null ? undefined : agentMessageForFirstSolution;
        this.agentMessageForSecondSolution = agentMessageForSecondSolution === null ? undefined : agentMessageForSecondSolution;
    }
}